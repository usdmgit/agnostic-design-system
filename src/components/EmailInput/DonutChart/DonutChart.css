@import '../../styles/theme/index.css';

.donutchart-full-circle {
  fill: var(--donutchart-full-circle-fill, transparent);
  stroke: var(--donutchart-full-circle-stroke, var(--structure-color-100));
}

.donutchart-indicator {
  fill: var(--donutchart-indicator-fill, transparent);
  stroke: var(--donutchart-indicator-stroke, #4c7ef6);
  stroke-dasharray: var(--donutchart-indicator-stroke-dasharrray, 0 10000);
  transition: var(--donutchart-indicator-transition, stroke-dasharray 0.3s ease);
}

.donutchart {
  margin: var(--donutchart-margin, 0 auto);
  border-radius: var(--donutchart-border-radius, 50%);
  display: var(--donutchart-display, block);
}

.donutchart-text {
  align-items: var(--donutchart-text-align-items, center);
  display: var(--donutchart-text-display, flex);
  flex-direction: var(--donutchart-text-flex-direction, column);
  justify-content: var(--donutchart-text-justify-content, center);
  font-family: var(--donutchart-text-font-family, 'Inter');
  fill: var(--donutchart-text-fill, --structure-color-900);
  position: var(--donutchart-text-position, absolute);
  left: var(--donutchart-text-left, 0);
  top: var(--donutchart-text-top, 50%);
  bottom: var(--donutchart-text-bottom, 50%);
  width: var(--donutchart-text-width, 100%);
}

.donutchart-text-val {
  font-size: var(--donutchart-text-val-font-size, 22px);
}

.donutchart-text-label {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: var(--donutchart-text-label-gap, 5px);
  font-size: var(--donutchart-text-label-font-size, 12px);
  margin-top: var(--donutchart-text-label-margin-top, 5px);
}
