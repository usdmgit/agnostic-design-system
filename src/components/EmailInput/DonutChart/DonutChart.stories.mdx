# DonutChart Component

## Type: Simple

## DonutChart Props

|        Prop        |      Type       |                                                      Description                                                       | Default | Mandatory |
| :----------------: | :-------------: | :--------------------------------------------------------------------------------------------------------------------: | :-----: | :-------: |
|        size        |     number      | Specifies the overal size of the Donut Chart. We consider this number to calculate the chart radius and circunference. |    -    |   true    |
|    strokewidth     |     number      |                        Specifies the width of the border you want to have on your donut chart.                         |    -    |   true    |
|     completion     |     number      |                        Receives the percentage of the filled portion we will show on the chart                         |    -    |   true    |
|      message       |     string      |                             Receives the label that should be displayed below the number.                              |    -    |   false   |
|   iconForMessage   | React.ReactNode |                Receives the icon that would be prepended in the label. You can customize your own icon.                |    -    |   false   |
| variablesClassName |     string      |                               Receives a custom class name for style/css customization.                                |    -    |   false   |

## Getting Started

```javascript
import React, { useState } from 'react';
import { DonutChart } from '@codelittinc/agnostic-design-system';

const DonutChart = args => {
  return <DonutChart {...args} />;
};

export default DonutChart;
```

## CSS Variables Available

|        Property        |    Attribute     | State |
| :--------------------: | :--------------: | :---: |
| donutchart-full-circle |       fill       |       |
| donutchart-full-circle |      stroke      |       |
|  donutchart-indicator  |       fill       |       |
|  donutchart-indicator  |      stroke      |       |
|  donutchart-indicator  | stroke-dasharray |       |
|  donutchart-indicator  |    transition    |       |
|       donutchart       |      margin      |       |
|       donutchart       |  border-radius   |       |
|       donutchart       |     display      |       |
|    donutchart-text     |   align-items    |       |
|    donutchart-text     |      bottom      |       |
|    donutchart-text     |     display      |       |
|    donutchart-text     |  flex-direction  |       |
|    donutchart-text     |       fill       |       |
|    donutchart-text     |   font-family    |       |
|    donutchart-text     | justify-content  |       |
|    donutchart-text     |       left       |       |
|    donutchart-text     |     position     |       |
|    donutchart-text     |       top        |       |
|    donutchart-text     |      width       |       |
|  donutchart-text-val   |    font-size     |       |
| donutchart-text-label  |    font-size     |       |
| donutchart-text-label  |       gap        |       |
