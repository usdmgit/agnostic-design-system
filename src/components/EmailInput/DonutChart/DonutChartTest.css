.custom-donut-chart {
  --donutchart-full-circle-stroke: pink;
  --donutchart-indicator-stroke: red;
  --donutchart-text-font-family: 'Nunito Sans';
  --donutchart-text-val-font-size: 15px;
  --donutchart-text-label-font-size: 15px;
}
