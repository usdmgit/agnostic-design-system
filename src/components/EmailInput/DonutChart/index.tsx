import React from 'react';
import styles from '@/components/DonutChart/DonutChart.css';
import classnames from 'classnames';

interface Props {
  size: number;
  strokewidth: number;
  completion: number;
  message?: string;
  iconForMessage?: React.ReactNode;
  variablesClassName?: string;
}

const DonutChart = (props: Props) => {
  const { size, strokewidth, completion, message, iconForMessage, variablesClassName } = props;

  const halfsize = size * 0.5;
  const radius = halfsize - strokewidth * 0.5;
  const circumference = 2 * Math.PI * radius;
  const strokeval = (completion * circumference) / 100;
  const dashval = strokeval + ' ' + circumference;

  const trackstyle = { strokeWidth: strokewidth };
  const indicatorstyle = { strokeWidth: strokewidth, strokeDasharray: dashval };
  const rotateval = 'rotate(-90 ' + halfsize + ',' + halfsize + ')';

  return (
    <div style={{ position: 'relative' }}>
      <svg width={size} height={size} className={classnames('donutchart', variablesClassName)}>
        <circle
          r={radius}
          cx={halfsize}
          cy={halfsize}
          transform={rotateval}
          style={trackstyle}
          className={styles['donutchart-full-circle']}
        />
        <circle
          r={radius}
          cx={halfsize}
          cy={halfsize}
          transform={rotateval}
          style={indicatorstyle}
          className={styles['donutchart-indicator']}
        />
        <text className={styles['donutchart-text']} x={halfsize} y={halfsize} />
      </svg>
      <span className={styles['donutchart-text']}>
        <span className={styles['donutchart-text-val']}>{completion}</span>
        {message && (
          <span className={styles['donutchart-text-label']}>
            {iconForMessage} {message}
          </span>
        )}
      </span>
    </div>
  );
};

export default DonutChart;
