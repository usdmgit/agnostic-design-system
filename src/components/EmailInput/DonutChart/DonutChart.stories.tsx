import React from 'react';
import DonutChart from '@/components/DonutChart';
import { Meta } from '@storybook/react';
import mdx from './DonutChart.stories.mdx';
import classNames from 'classnames';

import styles from './DonutChartTest.css';

export default {
  title: 'Components/DonutChart',
  component: DonutChart,
  parameters: {
    docs: {
      page: mdx
    }
  }
} as Meta;

const Template = args => {
  return <DonutChart {...args} />;
};

export const Default = Template.bind({});
Default.args = {
  completion: 44,
  message: 'No Risk',
  iconForMessage: <span> &#8861; </span>,
  size: 180,
  strokewidth: 10
};

export const CustomStyles = Template.bind({});
CustomStyles.args = {
  completion: 55,
  message: 'Completed',
  size: 200,
  strokewidth: 15,
  variablesClassName: classNames(styles['custom-donut-chart'])
};
