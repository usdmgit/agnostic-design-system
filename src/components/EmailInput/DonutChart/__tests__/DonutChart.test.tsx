import React from 'react';
import DonutChart from '@/components/DonutChart';
import { mount } from 'enzyme';

describe('DonutChart', () => {
  it('renders correctly', () => {
    const args = {
      completion: 55,
      message: 'Completed',
      size: 150,
      strokewidth: 10
    };
    const wrapper = mount(<DonutChart {...args} />);
    expect(wrapper).toMatchSnapshot();
  });
});
