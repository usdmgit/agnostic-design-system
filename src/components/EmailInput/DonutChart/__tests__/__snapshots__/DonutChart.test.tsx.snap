// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonutChart renders correctly 1`] = `
<DonutChart
  completion={55}
  message="Completed"
  size={150}
  strokewidth={10}
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <svg
      className="donutchart"
      height={150}
      width={150}
    >
      <circle
        className="donutchart-full-circle"
        cx={75}
        cy={75}
        r={70}
        style={
          Object {
            "strokeWidth": 10,
          }
        }
        transform="rotate(-90 75,75)"
      />
      <circle
        className="donutchart-indicator"
        cx={75}
        cy={75}
        r={70}
        style={
          Object {
            "strokeDasharray": "241.90263432641405 439.822971502571",
            "strokeWidth": 10,
          }
        }
        transform="rotate(-90 75,75)"
      />
      <text
        className="donutchart-text"
        x={75}
        y={75}
      />
    </svg>
    <span
      className="donutchart-text"
    >
      <span
        className="donutchart-text-val"
      >
        55
      </span>
      <span
        className="donutchart-text-label"
      >
         
        Completed
      </span>
    </span>
  </div>
</DonutChart>
`;
